src/ServerScriptService/Terrain/Generator/Beacon.lua

```diff
@@ -2,14 +2,41 @@ local Grid2D = require(script.Parent.Parent.Grid2D)
 
 local Beacon = {}
 
-function Beacon.Generate(islandConfig, beaconConfig)
-	return Grid2D.new() -- a grid of beacon positions -> true
-	-- this is some very simple math. simply add a beacon to the center, then the additional beacons are half way out from the island, in a circular pattern.
-	-- be sure to not add decimals to the generator - rounding is important
+function Beacon.Generate(beaconConfig)
+	local beaconGrid = Grid2D.new()
+
+	if beaconConfig.BeaconCount == 0 then
+		return beaconGrid
+	end
+
+	local beaconCount = beaconConfig.BeaconCount - 1
+
+	beaconGrid:Set(0, 0, true)
+	local beaconAngleDifference = 360 / beaconCount
+	local beaconAngle = 0
+
+	for y = 0, beaconCount do
+		local angleCos = math.cos(math.rad(beaconAngle))
+		local angleSin = math.sin(math.rad(beaconAngle))
```

Comment:
'angleCos' and 'angleSin' are misleading. The sin and cos function do not return angles - they return displacements. I suggest using 'x' and 'z' as variable names here.